### ISSUE TITLE
Update web "howto" to use endpoints and better TLS.

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8596]
|[<img alt="Lukasa's avatar" src="https://avatars.githubusercontent.com/u/1382556?s=50" width="50" height="50">](https://github.com/Lukasa)| @Lukasa reported|
|-|-|
|Trac ID|trac#8596|
|Type|enhancement|
|Created|2016-07-13 09:23:45Z|
|Branch|https://github.com/twisted/twisted/tree/8596-lukasa-webhowtoendpoints|

The web howto docs use {connect,listen}SSL, which means that they rely on ContextFactory, which we're trying to deprecate (see [#6923](https://github.com/twisted/twisted/issues/6923)). We should change all these docs to use endpoints instead, which resolves this problem.

<details><summary>Searchable metadata</summary>

```
trac-id__8596 8596
type__enhancement enhancement
reporter__Lukasa Lukasa
priority__normal normal
milestone__None None
branch__8596_lukasa_webhowtoendpoints 8596-lukasa-webhowtoendpoints
branch_author__ 
status__closed closed
resolution__fixed fixed
component__web web
keywords__None None
time__1468401825344788 1468401825344788
changetime__1468485367805569 1468485367805569
version__None None
owner__Lukasa Lukasa

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="Lukasa's avatar" src="https://avatars.githubusercontent.com/u/1382556?s=50" width="50" height="50">](https://github.com/Lukasa)<a name="note_1"></a>|@Lukasa commented|
|-|-|

In changeset e73856b92b2dcc9990c351b31afff50d2159a97f
```
#!CommitTicketReference repository="" revision="e73856b92b2dcc9990c351b31afff50d2159a97f"
Add topfile for #8596.
```

**Comment by twisted-trac** [comment]
|[<img alt="Lukasa's avatar" src="https://avatars.githubusercontent.com/u/1382556?s=50" width="50" height="50">](https://github.com/Lukasa)<a name="note_2"></a>|@Lukasa commented|
|-|-|

A patch is available at https://github.com/twisted/twisted/pull/324

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_3"></a>|@adiroiban set owner to @Lukasa|
|-|-|

Thanks. Looks good.

Only a minor comment regarding making explicit that endpoints are also for connect, not only for listen.

Please review my inline comment and merge at will :)

Thanks again!

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|GitHub set status to `closed`|
|-|-|

In changeset dea7cd40f5ddd1c199f003c55a81f0a97a98ec86
```
#!CommitTicketReference repository="" revision="dea7cd40f5ddd1c199f003c55a81f0a97a98ec86"
Merge 8596-lukasa-webhowtoendpoints: Change the web howto to use endpoints

Author: lukasa
Reviewer: adiroiban
Fixes: #8596
```
